src/react/components/GenericForm.jsx

```diff
@@ -10,120 +11,226 @@ import 'antd/es/input/style/index.js';
-                    ) }
-                </Select>
-            );
-        case 'submit':
+    /**
+     * 
+     * @param {Object} fields - array containing all fields
+     * @returns {React.Component} 
+     */
+    createFormItems (fields, formSettings) {
+        return fields.map((field) => {
             return (
-                <Button type="primary" htmlType={ field.type }>
-                    { field.value }
-                </Button>
+                <StyledFormItem
+                    key={ field.name }
+                    name={ field.type !== 'buttongroup' ? field.name : '' }
+                    label={ formSettings.showLabels ? field.label : '' }
+                    rules={ field.rules }
+                    initialValue={ this._getFieldInitialValue(field) }
+                >
+                    { this.createFormInput( field ) }
+                </StyledFormItem>
             );
-        default:
+        });
+    }
+
+    /**
+     * Create single Form.Item content with provided field properties
+     * 
+     * @param {Object} field              - object containing information for single field
+     * @param {String} field.type         - field type as string {text / textarea / info / dropdown}
+     * @param {String} field.placeholder  - placeholder text for the current field
+     * @param {String|Number} field.value - value for current field used in info card / drowdown / submit button
+     * 
+     * @returns {Component} React component for the provided field
+     */
+
+    createFormInput (field) {
+        if (!field) {
             return null;
+        }
+    
+        const fieldKey = field.name + '_' + field.type + '_field';
+    
+        switch(field.type) {
+            case 'text':
+                return (
+                    <Input
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                    />
+                );
+            case 'textarea':
+                return (
+                    <TextArea
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                    />
+                );
+            case 'info':
+                return (
+                    <Card key={ fieldKey }>
+                        { field.value }
+                    </Card>
+                );
+            case 'dropdown':
+                return (
+                    <Select
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                        dropdownStyle={{ zIndex: zIndexValue }}
+
+                    >
+                        { field.value.map((singleOption) => {
+                            return (
+                                <Select.Option
+                                    value={ singleOption.value }
+                                    key={ singleOption.label || singleOption.value }
```

Comment:
We fallback to support non-object options for label. We should do it here as well.
```suggestion
                                    value={ singleOption.value || singleOption }
                                    key={ singleOption.label || singleOption.value || singleOption }
```